app/customer.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+import math
+
+from app.car import Car
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(self, customer_data: dict) -> None:
+        self.name: str = customer_data.get("name")
+        self.product_cart: dict = customer_data.get("product_cart")
+        self.location: list[int] = customer_data.get("location")
+        self.money: float | int = customer_data.get("money")
+        self.car: Car = Car(customer_data.get("car"))
+
+    def calculate_distance(self, shop_location: list[int]) -> float:
+        return math.sqrt(
+            (self.location[0] - shop_location[0]) ** 2
+            + (self.location[1] - shop_location[1]) ** 2
+        ) * 2

```

Comment:
The multiplication by 2 in the `calculate_distance` method seems unnecessary unless the task explicitly requires doubling the distance. Please verify the task requirements.